File: Queue.c
#include<stdio.h>
#include<stdlib.h>

struct node
{
    int data;
    struct node * next;

};


typedef struct node NODE;
typedef struct node * PNODE;
typedef struct node ** PPNODE;

void EnQueue(PPNODE Head,int no)
{
    PNODE newn=NULL;
    newn=(PNODE)malloc(sizeof(NODE));
    PNODE temp=*Head;

    newn->data=no;
    newn->next=NULL;


    if(*Head==NULL)
    {
        *Head=newn;
    }
    else
    {
       while (temp->next!=NULL)
       {
           temp=temp->next;
       }

       temp->next=newn;
       
    }

}


void Dqueue(PPNODE Head)
{
    PNODE temp=*Head;
    int no=0;
    
    if(*Head==NULL)
    {
        return;
    }

    else
    {
        
        temp=*Head;
        no=temp->data;
        *Head=temp->next;
        free(temp);
        printf("Removed data is:%d\n",no);
        
    }
}


void Display(PNODE head)
{
    printf("Elemnts from queue are \n");
    while(head!=NULL)
    {
        printf("|%d|->",head->data);
        head=head->next;
    }
    printf("NULL\n");
}

int Count(PNODE head)
{
    int iCnt=0;
    while(head!=NULL)
    {
        iCnt++;
        head=head->next;
    }

    return iCnt;
}



int main()
{
   PNODE First=NULL;
   int iRet=0;

   EnQueue(&First,11);
   EnQueue(&First,21);
   EnQueue(&First,51);

   Display(First);

   
   iRet=Count(First);
   printf("Count is:%d",iRet);



   Dqueue(&First);

   Display(First);



    return 0;
}

File: Queue.java
import java.lang.*;
import java.util.*;

class node {
    public int data;
    public node next;
}

class Queuee {

    public node Head;
    public int Count;

    public Queuee() {
        Head = null;
        Count = 0;
    }

    public void Display() {
        node temp = Head;
        System.out.print("Elements from queue are:");
        while (temp != null) {
            System.out.print("|" + temp.data + "|->");
            temp = temp.next;
        }
        System.out.println("NULL");
    }

    public int CountNode() {
        return Count;
    }

    public void Enqueue(int no) {
        node newn = new node();

        newn.data = no;
        newn.next = null;

        if (Head == null) {
            Head = newn;
        } else {
            node temp = Head;
            while (temp.next != null) {
                temp = temp.next;
            }
            temp.next = newn;
        }

        Count++;
    }

    public void Dequeue() {
        if (Head == null) {
            System.out.println("Queue is empty");
            return;
        }

        else {

            node temp = Head;
            int no = temp.data;
            Head = temp.next;
            temp = null;
            System.out.println("removed data is " + no);

        }
        Count--;
    }

}

class Queue {
    public static void main(String args[]) {
        Queuee obj1 = new Queuee();

        obj1.Enqueue(11);
        obj1.Enqueue(21);
        obj1.Enqueue(51);
        obj1.Enqueue(101);

        obj1.Display();

        System.out.println("Count of elements are :" + obj1.CountNode());

        obj1.Dequeue();
        obj1.Dequeue();
        obj1.Dequeue();

        obj1.Display();

        System.out.println("Count of elements are :" + obj1.CountNode());

        obj1.Dequeue();
        obj1.Dequeue();

    }
}